internal/csi/csi.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package csi contains support for processing the information in a CSI file.
+package csi
+
+import (
+	"github.com/googlegenomics/htsget/internal/genomics"
+)
+
+// RegionContainsBin indicates if the given region contains the bin described by
+// referenceID and binID.
+func RegionContainsBin(region genomics.Region, referenceID int32, binID uint32, bins []uint16) bool {
+	if region.ReferenceID >= 0 && referenceID != region.ReferenceID {
+		return false
+	}
+
+	if region.Start == 0 && region.End == 0 {
+		return true
+	}
+
+	for _, id := range bins {
+		if uint32(id) == binID {
+			return true
+		}
+	}
+	return false
+}
+
+// BinsForRange returns the list of bins that may overlap with the zero-based region
+// defined by [start, end). The minShift and depth parameters control the minimum interval width
+// and number of binning levels, respectively.
+// This function is derived from the C examples in the CSI index specification.
```

Comment:
Let's move this last line to just before the bin calculation because (a) that's the part that's derived and (b) we don't want this to appear in the public documentation since it's an implementation comment not a usage comment.

Reply:
done